examples/robot_example.cpp

```diff
@@ -0,0 +1,94 @@
+
+// #include "kodlab_mjbots_sdk/robot_control_loop.h"
+// #include "kodlab_mjbots_sdk/behavior.h"
+// #include "kodlab_mjbots_sdk/robot_super.h"
+#include <vector>
+#include <iostream>
+#include <memory>
+#include "kodlab_mjbots_sdk/common_header.h"
+#include "ManyMotorLog.hpp"
+#include "ModeInput.hpp"
+#include "kodlab_mjbots_sdk/robot_interface.h"
+#include "kodlab_mjbots_sdk/mjbots_control_loop.h"
+
+#include "examples/simple_robot.h"
+
+class SimpleRobotControlLoop : public kodlab::mjbots::MjbotsControlLoop<ManyMotorLog, ModeInput, SimpleRobot>
+{
+    using MjbotsControlLoop::MjbotsControlLoop;
+
+    void CalcTorques() override
+    {
+        robot_->Update();
+    }
+    void PrepareLog() override
+    {
+        for (int servo = 0; servo < num_motors_; servo++)
+        {
+            log_data_.positions[servo] = robot_->GetJointPositions()[servo];
+            log_data_.velocities[servo] = robot_->GetJointVelocities()[servo];
+            log_data_.modes[servo] = static_cast<int>(mjbots_interface_->GetJointModes()[servo]);
+            log_data_.torques[servo] = robot_->GetJointTorqueCmd()[servo];
+        }
+        for (int servo = num_motors_; servo < 13; servo++)
+        {
+            log_data_.positions[servo] = 0;
+            log_data_.velocities[servo] = 0;
+            log_data_.modes[servo] = 0;
+            log_data_.torques[servo] = 0;
+        }
+    }
+
+    void ProcessInput() override
+    {
+        robot_->mode = lcm_sub_.data_.mode;
+        std::cout << "Switching to behavior " << robot_->mode << std::endl;
+        // If the kill robot mode is detected kill robot using CTRL_C flag handler.
+        if (robot_->mode == robot_->KILL_ROBOT)
+        { // KILL_ROBOT
+            kodlab::CTRL_C_DETECTED = true;
+        }
+    }
+};
+
+template <class joint_type>
```

Comment:
This class template could be moved to the joint base class probably though templating would get weird. Useful way to used shared pointers without forcing c++syntax. Probably a few other functions that could be moved in here like the getters and stuff, but right now this is implicitly converted to a vector of shared pointers.

Reply:
I thought about deriving std::vector but decided it was a bad idea with lots of ways it could go wrong.

Reply:
This should definitely move out of `robot_example`.  I don't think moving to the `JointBase` class makes sense as it is a vector composing one or several `JointBase`'s.  I would just keep this as-is, but stick it in its own class (perhaps in the `joint_base.h` header but outside of the `JointBase` class) to be included as needed.  The implicit conversion and wrapping of `std::shared_ptr` is nice for user interaction.

Reply:
I meant into the same header. My first thought was in something like the common_header but for joints and that is joint_base.h

Reply:
I meant into the same header. My first thought was in something like the common_header but for joints and that is joint_base.h

Reply:
Moved to joint_header

Reply:
Moved to joint_header